WeekFormulas.ts

```diff
@@ -203,27 +207,42 @@ function protectWeek(weekNum: number): void {
         wplProtection = getColumn(weekNum, 8).protect().setDescription("WPL PROTECTION");
     const sheetEditors = wahfProtection.getEditors();
 
-    wahfProtection.removeEditors(sheetEditors).addEditors(databaseAdmin);
+    //grab row length
+    
+    const proctectedSheet = SpreadsheetApp.getActiveSpreadsheet().getSheetByName("Protection");
+    const calculateSheet = SpreadsheetApp.getActiveSpreadsheet().getSheetByName("");
+    //grab row length
+
+    var range = getFormulaResult("=COUNTA(Protection!B1:1)");
+    const procDesk = proctectedSheet.getRange(1, 3, 1, range-1).getValues() as string[][];
+
+    range = getFormulaResult("=COUNTA(Protection!B2:2)");
+    const procStudy = proctectedSheet.getRange(2,  3, 1, range-1).getValues() as string[][];
+
+    range = getFormulaResult("=COUNTA(Protection!B3:3)");
+    const procSeminar = proctectedSheet.getRange(3, 3, 1, range-1).getValues() as string[][];
+
+    range = getFormulaResult("=COUNTA(Protection!B4:4)");
+    const procDbase = proctectedSheet.getRange(4, 3, 1, range-1).getValues() as string[][];
+
     semProtection
         .removeEditors(sheetEditors)
```

Comment:
You may be able to remove the `removeEditors` calls. On the next PR, we can work on when someone edits the protection sheet, that it should retroactively remove/add editors to past protections.